config/logger.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/utils/fileutils"
+)
+
+const (
+	LogRotateSize           = 10000
+	LogCompress             = true
+	LogRotateMaxAge         = 0
+	LogRotateMaxBackups     = 0
+	LogFilename             = "mattermost.log"
+	LogNotificationFilename = "notifications.log"
+	LogMinLevelLen          = 5
+	LogMinMsgLen            = 45
+	LogDelim                = " "
+	LogEnableCaller         = true
+)
+
+type fileLocationFunc func(string) string
+
+func MloggerConfigFromLoggerConfig(s model.LogSettings, configSrc LogConfigSrc, getFileFunc fileLocationFunc) (mlog.LoggerConfiguration, error) {
+	cfg := make(mlog.LoggerConfiguration)
+
+	var targetCfg mlog.TargetCfg
+	var err error
+
+	// add the simple logging config
+	if *s.EnableConsole {
+		targetCfg, err = makeSimpleConsoleTarget(*s.ConsoleLevel, *s.ConsoleJson, *s.EnableColor)
+		if err != nil {
+			return cfg, err
+		}
+		cfg["_defConsole"] = targetCfg
+	}
+
+	if *s.EnableFile {
+		targetCfg, err = makeSimpleFileTarget(getFileFunc(*s.FileLocation), *s.FileLevel, *s.FileJson)
+		if err != nil {
+			return cfg, err
+		}
+		cfg["_defFile"] = targetCfg
+	}
+
+	if configSrc == nil {
+		return cfg, nil
+	}
+
+	// add advanced logging config
+	cfgAdv := configSrc.Get()
+	cfg.Append(cfgAdv)
+
+	return cfg, nil
+}
+
+func MloggerConfigFromAuditConfig(auditSettings model.ExperimentalAuditSettings, configSrc LogConfigSrc) (mlog.LoggerConfiguration, error) {
+	cfg := make(mlog.LoggerConfiguration)
+
+	var targetCfg mlog.TargetCfg
+	var err error
+
+	// add the simple audit config
+	if *auditSettings.FileEnabled {
+		targetCfg, err = makeSimpleFileTarget(*auditSettings.FileName, "error", true)
+		if err != nil {
+			return cfg, err
+		}
+
+		// apply audit specific levels
+		targetCfg.Levels = []mlog.Level{mlog.LvlAuditAPI, mlog.LvlAuditContent, mlog.LvlAuditPerms, mlog.LvlAuditCLI}
+
+		// apply audit specific formatting
+		targetCfg.FormatOptions = json.RawMessage(`{"disable_timestamp": true, "disable_msg": true, "disable_stacktrace": true, "disable_level": true}`)
+
+		cfg["_defAudit"] = targetCfg
+	}
+
+	if configSrc == nil {
+		return cfg, nil
+	}
+
+	// add advanced audit config
+	cfgAdv := configSrc.Get()
+	cfg.Append(cfgAdv)
+
+	return cfg, nil
+}
+
+// DON'T USE THIS Modify the level on the app logger
+func DisableDebugLogForTest(testLogger *mlog.Logger) {
+	// TODO: allow mlog to change log levels
+}
+
+// DON'T USE THIS Modify the level on the app logger
+func EnableDebugLogForTest(testLogger *mlog.Logger) {
+	// TODO: allow mlog to change log levels
+}
```

Comment:
Agree completely.  I left this in because I wasn't sure if any tests relied on less debug level logs.  I think it's safe to remove these now and use a better mechanism to set log levels before running tests.

Reply:
All `DisableDebugLogForTest` calls removed via https://github.com/mattermost/mattermost-server/pull/18118/commits/688310dca8884819fe31d941ba80d618a00fdaba

Reply:
Awesome, sounds good!